### PR TITLE
Fix History erasing all changes error

### PR DESCRIPTION
**Pull request by jparisu**
For unkown reasons RTPS methods show warnings.
Fix: Do not use them.

Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

File: include/ddsrouter/communication/PayloadPool.hpp [hunk 1]
```diff
@@ -158,6 +158,13 @@ class PayloadPool : public fastrtps::rtps::IPayloadPool
 
 protected:
 
+    bool reserve_(
```
Comment (juanlofer-eprosima) [inline include/ddsrouter/communication/PayloadPool.hpp:161]: Returned bool is used nowhere, make it void or make use of it.

File: src/cpp/communication/CopyPayloadPool.cpp [hunk 2]
```diff
@@ -74,11 +70,7 @@ bool CopyPayloadPool::get_payload(
 bool CopyPayloadPool::release_payload(
         Payload& payload)
 {
-    get_payload(0, payload);
-    payload.length = 0;
-    payload.pos = 0;
-    payload.max_size = 0;
-    payload.data = nullptr;
+    release_(payload);
```
Comment (juanlofer-eprosima) [inline src/cpp/communication/CopyPayloadPool.cpp:73]:
```suggestion
    return release_(payload);
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: src/cpp/communication/CopyPayloadPool.cpp [hunk 3]
```diff
@@ -31,41 +31,37 @@ bool CopyPayloadPool::release_payload(
...
-    if (size > 0)
-    {
-        // Set that a payload has been reserved
-        add_reserved_payload_();
-    }
-    else
-    {
-        // Set that a payload has been released
-        add_release_payload_();
-    }
```
Comment (juanlofer-eprosima) [inline src/cpp/communication/CopyPayloadPool.cpp:59]: Should we keep this just in case? Although should never happen.

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: include/ddsrouter/communication/PayloadPool.hpp (hunk 1 above)
Comment (jparisu) [reply include/ddsrouter/communication/PayloadPool.hpp:161]: Actually reserve could fail, but `reserve` method does not return return value in Fast DDS, so I would say the issue is there and not here.
I understand it is useless, but I would leave it this way. It is better to have it useless than change the API in the future.

File: src/cpp/communication/CopyPayloadPool.cpp (hunk 2 above)
Comment (jparisu) [reply src/cpp/communication/CopyPayloadPool.cpp:73]: :+1:

File: src/cpp/communication/CopyPayloadPool.cpp (hunk 3 above)
Comment (jparisu) [reply src/cpp/communication/CopyPayloadPool.cpp:59]: This was my fault, as I did not realize that method `empty` actually release a Payload. (How could I? The name of the method is so descriptive, and it so well commented...).
Thus, it should never call reserve with size 0 (actually this is a good check to add).

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**